aws/resource_aws_lambda_layer_version_permission.go

```diff
@@ -0,0 +1,211 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsLambdaLayerVersionPermission() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLambdaLayerVersionPermissionAdd,
+		Read:   resourceAwsLambdaLayerVersionPermissionGet,
+		Delete: resourceAwsLambdaLayerVersionPermissionRemove,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"layer_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"layer_version": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"statement_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"principal": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"organization_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"revision_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsLambdaLayerVersionPermissionAdd(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	layerArn := aws.String(d.Get("layer_arn").(string))
+	layerVersion := aws.Int64(int64(d.Get("layer_version").(int)))
+	statementId := aws.String(d.Get("statement_id").(string))
+	principal := aws.String(d.Get("principal").(string))
+	organizationId, hasOrganizationId := d.GetOk("organization_id")
+	action := aws.String(d.Get("action").(string))
+
+	params := &lambda.AddLayerVersionPermissionInput{
+		LayerName:     layerArn,
+		VersionNumber: layerVersion,
+		Action:        action,
+		Principal:     principal,
+		StatementId:   statementId,
+	}
```

Comment:
can the above vars be inside the `AddLayerVersionPermissionInput` block?

like so:
```
	params := &lambda.AddLayerVersionPermissionInput{
		LayerName:     aws.String(d.Get("layer_arn").(string)),
... other arguments
	}
```

Reply:
vars moved inside